blog/content/post/2022-06-17-fobnail_minimal_os.md

```diff
@@ -0,0 +1,253 @@
+system image should be available in `build/tmp/deploy/images/fobnail-machine/`.
+It was prepared to run from SD card. We need to use `bmaptool` to flash image on
+the card:
+
+```
+$ bmaptool copy --bmap fobnail-base-image-debug-fobnail-machine.wic.bmap \
+    fobnail-base-image-debug-fobnail-machine.wic.gz /dev/sdX
+```
+
+> Note: We recommend to use debug version as it is passwordless.
+
+We have also published a
+[document](https://fobnail.3mdeb.com/meta-fobnail-in-dlme/) outlining our
+efforts to make our minimal OS run in DLME. We mentioned that the components
+from TrenchBoot were used, but in practice, using them on the PC Engines apu2
+platform was not trivial.
+
+### Attestation workflow
+
+The minimum operating system we have prepared includes a set of functionalities
+that allows to test the following scenario. Imagine we got a Fobnail Token along
+with a minimal OS. We can now run it on our platform and validate it. If we get
+a positive result, we have the green light to boot our target system. Otherwise,
+unfortunately, we receive information that something may have changed on our
+platform and it is better not to boot anything else. The diagram below shows the
+whole situation.
+
+![Fobnail Token flow with minimal OS](/img/ft-minimal-os.png)
+
+In order to run the test we need to perform following steps. In our scenario we
+boot to the minimal OS, run attestation and finally boot `Ubuntu 20.04 LTS Live`
+flashed on pendrive.
+
+1. Boot [meta-fobnail](https://github.com/fobnail/meta-fobnail) image in DLME -
+   this was described [here](https://fobnail.3mdeb.com/meta-fobnail-in-dlme/).
+
+2. Log into the platform, the server should be started on boot, to see logs,
+   please run the following command.
+
+```
+# journalctl -fu fobnail-attester
+-- Journal begins at Tue 2022-06-07 15:04:57 UTC. --
+Jun 07 15:05:14 tb systemd[1]: Started Fobnail Attester service.
+```
+
+3. Connect Fobnail Token to PC Engines apu2. If the device is detected properly
+   system should print following information
+
+```
+[   42.108151] usb 2-2: new full-speed USB device number 2 using xhci_hcd
+[   42.334313] cdc_eem 2-2:1.0 usb0: register 'cdc_eem' at usb-0000:00:10.0-2, CDC EEM Device, 5e:f9:bb:9b:dd:06
+[   42.355332] usbcore: registered new interface driver cdc_eem
+```
+
+4. Wait a few seconds for provisioning and attestation. During this procedure
+   `Fobnail Attester` should print information about received data
+
+```
+Received message: ek
+Received message: aik
+Received message: challenge
+Received message: metadata
+MAC:  0: D:B9:53:D2:50
+SMBIOSv2
+Manufacturer >PC Engines<
+Product Name >apu2<
+Serial Number >1373268<
+Received message: rim
+Received message: metadata
+MAC:  0: D:B9:53:D2:50
+SMBIOSv2
+Manufacturer >PC Engines<
+Product Name >apu2<
+Serial Number >1373268<
+Received message: quote
+```
+
+5. Attestation was finished successfully when green LED blinked on Fobnail
```

Comment:
Blinking reports successful provisioning, attestation result is reported by a constant light for 10 seconds.

Reply:
changed https://github.com/3mdeb/news-and-ideas/pull/496/commits/acea6f8ce69e7cd420e09461d14ba223d9597e59#diff-5f2030137d33048b104fbebb6f1f3ce21c3e5522e57c8841c0fc8be5ceb88e2cL217

Reply:
changed sentence: https://github.com/3mdeb/news-and-ideas/pull/496/commits/9ce86f0f8c8e68faa0fb97df8cf07865b921ea22#diff-470addb504c37cdd533976b2247167df2c35b0e4b11b68ffa5c03fea7ffcd820L217